src/openrct2/world/Map.cpp

```diff
@@ -1539,6 +1542,92 @@ void map_update_tiles()
     }
 }
 
+void map_calc_highest_tile_height(bool fullCheck)
+{
+    int32_t maxTileIndex = MAXIMUM_MAP_SIZE_TECHNICAL * MAXIMUM_MAP_SIZE_TECHNICAL;
+    int32_t tilesToCheck = maxTileIndex;
+
+    // If we're doing a full check of the map then zero out the loop
+    // position to start at tile zero, checking all tiles before returning.
+    // Otherwise check a limited number of tiles per each time the
+    // function is called.
+    if (fullCheck)
+    {
+        gMapHighestTileHeightLoopPosition = 0;
+    }
+    else
+    {
+        tilesToCheck /= 128;
```

Comment:
Dividing the check across 128 cycles of the game loop when not performing a full check. This seemed performant but might need some thought on what the right number here is.

Reply:
The current time it needs to complete a full cycle: (256 * 256) / 128 * 40 = 20480 ms. Also I'm not sure if it should always  run, when nothing changes then it doesn't have to do anything.

Reply:
You could have a flag that gets invalidated from the game actions to retrigger the checks.

Reply:
That makes sense to me, I'll add the flag, and a function to invalidate. I'll then place the invalidate call in actions that modify elements.